src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java

```diff
@@ -372,4 +386,55 @@ public void deleteInventory(MedicalInventory medicalInventory) throws OHServiceE
 		}
 		ioOperations.deleteInventory(medicalInventory);
 	}
+
+	/**
+	 * Confirm the Inventory rows of inventory.
+	 *
+	 * @param inventory - The {@link MedicalInventory}
+	 * @param inventoryRowSearchList- The list of {@link MedicalInventory}
+	 * @throws OHDataValidationException
+	 */
+	public void confirmMedicalInventoryRow(MedicalInventory inventory, List<MedicalInventoryRow> inventoryRowSearchList) throws OHServiceException {
+		// validate the inventory
+		this.validateMedicalInventoryRow(inventory, inventoryRowSearchList);
+		// create Movements
+		String dischargeCode = inventory.getDischargeType();
+		String chargeCode = inventory.getChargeType();
+		Integer supplierId = inventory.getSupplier();
+		String wardCode = inventory.getDestination();
+		MovementType chargeType = medicalDsrStockMovementTypeBrowserManager.getMovementType(chargeCode);
+		MovementType dischargeType = medicalDsrStockMovementTypeBrowserManager.getMovementType(dischargeCode);
+		Supplier supplier = supplierManager.getByID(supplierId);
+		Ward ward = wardManager.findWard(wardCode);
+		LocalDateTime today = LocalDateTime.now();
+		for (Iterator<MedicalInventoryRow> iterator = inventoryRowSearchList.iterator(); iterator.hasNext();) {
+			String reference = "Ref-" + LocalDateTime.now();
```

Comment:
Shouldn't the reference be given by the inventory record itself?

```suggestion
			String reference = "Ref-" + LocalDateTime.now();
```

Reply:
here is the reference of movement not the reference of inventory

Reply:
I know, but why should they differ?

Reply:
okay I will use the same reference

Reply:
Look that method if we use the same reference for different movement : `protected List<OHExceptionMessage> checkReferenceNumber(String referenceNumber) throws OHServiceException {
		List<OHExceptionMessage> errors = new ArrayList<>();
		if (referenceNumber == null || referenceNumber.isEmpty()) {
			errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.medicalstock.multiplecharging.pleaseinsertareferencenumber.msg")));
		} else {
			if (refNoExists(referenceNumber)) {
				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.medicalstock.multiplecharging.theinsertedreferencenumberalreadyexists.msg")));
			}
		}
		return errors;
	}`

Reply:
we always do that, just use the methods:

- newMultipleChargingMovements
- newMultipleDischargingMovements

Reply:
okay